packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -52,24 +52,50 @@ export class Wallet {
             channelId: channelResult.channelId,
             currentStatus: newObjective.status,
             objectiveId: newObjective.objectiveId,
-            done: this.ensureObjective(newObjective, getMessages(createResult)).catch(error => ({
-              type: 'InternalError' as const,
-              error,
-            })),
+            done: this.ensureObjective(newObjective, getMessages(createResult)),
           };
         } catch (error) {
+          // TODO: This is slightly hacky but it's less painful then having to narrow the type down every time
+          // you get a result back from the createChannels method
+          // This should be looked at in https://github.com/statechannels/statechannels/issues/3461
           return {
-            channelParameters: p,
+            channelId: 'ERROR',
+            currentStatus: 'failed' as const,
+            objectiveId: 'ERROR',
             done: Promise.resolve({type: 'InternalError' as const, error}),
           };
         }
       })
     );
   }
 
+  /**
+   * Finds any approved objectives and attempts to make progress on them.
+   * This is useful for restarting progress after a restart or crash.
+   * @returns A collection of ObjectiveResults. There will be an ObjectiveResult for each approved objective found.
+   */
+  public async jumpStartObjectives(): Promise<ObjectiveResult[]> {
+    const objectives = await this._engine.getApprovedObjectives();
+    const objectiveIds = objectives.map(o => o.objectiveId);
+    // Instead of getting messages per objective we just get them all at once
+    // This will prevent us from querying the database for each objective
+    const syncMessages = getMessages(await this._engine.syncObjectives(objectiveIds));
+    return Promise.all(
+      objectives.map(async o => {
+        return {
+          objectiveId: o.objectiveId,
+          currentStatus: o.status,
+          channelId: o.data.targetChannelId,
+          done: this.ensureObjective(o, syncMessages),
```

Comment:
`syncMessages` contains data for every approved objective. It's passed to `this.ensureObjective` once per active objective. What effect does this have?

Reply:
This means that we're sending out the `syncMessages` payload for every objective (which means a lot of redundant network requests).

This might be a case of premature optimization as I was trying to avoid calling `syncObjective` per objectives to prevent excess db queries.

Longer-term I think the `engine` should return messages indexed by `objectiveId` so it is easy to only send the correct messages. (#3461)

Reply:
Right. So, in the current system we need to choose between something like this list of options:
1. excessive # of DB calls
2. excessive network requests
3. modifying `syncObjectives` (#3461)
4. hacking the returned value of `syncObjectives`

If we're explicitly choosing the drawback (2), I think it's important to highlight this in the PR description.

Reply:
Actually, it's probably worth it to highlight this drawback in the code itself. What do you think?

Reply:
I agree! Done in 7758eb6